File: C_Problem_Sets/ps1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

//Cash money

int main(void)
{
    float change;
    do
    {
        change = get_float("How much change we owe you?\n");
    }
    while (change <= 0);

    //Prompts user to enter amount of change. Must be positive, otherwise the program will continue to ask.

    //program logic starts here.

    int cents = round(change * 100);

    int q, d, n, p;
    q = cents / 25;
    // printf("%i\n", q);
    cents = cents - q * 25;

    d = cents / 10;
    // printf("%i\n", d);
    cents = cents - d * 10;

    n = cents / 5;
    // printf("%i\n", n);
    cents = cents - n * 5;

    p = cents / 1;
    // printf("%i\n", p);
    int coins = q + d + n + p;
    printf("%i\n", coins);
}

File: Python_Problem_Sets/ps6/cash.py
from cs50 import get_int
from cs50 import get_float
from cs50 import get_string

while True:
    change = get_float("Change owed: ")
    if change > 0:
        break

change = 100 * change
Q = int(change / 25)
# print(Q)
# print(int(Q))
D = int((change - (Q * 25))/10)
N = int((change - (Q * 25) - (D * 10))/5)
P = int((change - (Q * 25) - (D * 10) - (N * 5)))

print(Q+D+N+P)
# print(f"{change}")
